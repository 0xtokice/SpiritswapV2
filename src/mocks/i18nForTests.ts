/* istanbul ignore next */
export const mockUseTranslation = () => ({
  t: key => key,
  i18n: {
    language: undefined,
  },
});
