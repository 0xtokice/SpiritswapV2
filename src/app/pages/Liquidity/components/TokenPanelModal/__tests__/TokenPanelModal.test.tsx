import { render } from '@testing-library/react';
import { ThemeWrapper } from 'app/components/shared/testing/ThemeWrapper';
import { mockUseTranslation } from 'mocks/i18nForTests';
import { TokenPanelModal } from '../index';

jest.mock('react-i18next', () => ({
  useTranslation: () => mockUseTranslation(),
}));

const token = {
  name: 'SpiritSwap Token',
  symbol: 'SPIRIT',
  address: '0x5Cc61A78F164885776AA610fb0FE1257df78E59B',
  chainId: 250,
  decimals: 18,
};
const amounts = {
  SPIRIT: 1000,
};
const Component = () => {
  return (
    <ThemeWrapper>
      <TokenPanelModal token={token} amounts={amounts} />
    </ThemeWrapper>
  );
};

describe('OptionsDropdown component', () => {
  test('renders without crashing', () => {
    const { container } = render(<Component />);
    expect(container.firstChild).toBeTruthy();
  });

  it('should render and match snapshot', () => {
    const { container } = render(<Component />);
    expect(container).toMatchSnapshot();
  });
});
