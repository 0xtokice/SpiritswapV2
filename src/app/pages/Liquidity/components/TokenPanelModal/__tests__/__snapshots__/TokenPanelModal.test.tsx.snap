// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionsDropdown component should render and match snapshot 1`] = `
.c0 {
  display: inline-grid;
  grid-auto-flow: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.5rem;
}

<div>
  <div
    class="chakra-stack css-19p7apo"
  >
    <div
      class="chakra-stack css-1rngsr6"
    >
      <p
        class="chakra-text css-1ylkl1b"
      >
        1,000
      </p>
      <div
        class="c0"
      >
        <div
          class="css-k008qs"
        >
          <img
            alt="SPIRIT logo"
            class="chakra-image__placeholder css-18pkb04"
            fallbackstrategy="onError"
            src="/images/tokens/default.png"
          />
        </div>
        <p
          class="chakra-text css-1xbhoz2"
        >
          SPIRIT
        </p>
      </div>
    </div>
    <div
      class="chakra-stack css-1vjp4pi"
    >
      <div
        class="css-k008qs"
      >
        <p
          class="chakra-text css-dferxh"
        >
          â‰ˆ 
          4,000
        </p>
        <p
          class="chakra-text css-14edhu4"
        >
          ( -0.1 )%
        </p>
      </div>
      <p
        class="chakra-text css-0"
      >
        common.tokenAmountPanel.balance: $1.97
      </p>
    </div>
  </div>
</div>
`;
